#482 enable docker build and fix docker module dependency cache issue (#506)
boozallen/aissemble: DRAFT_RELEASE_NOTES.md

# Breaking Changes
_Note: instructions for adapting to these changes are outlined in the upgrade instructions below._

>  - To improve the development cycle and docker build consistency, we have deprecated the docker_build() and local_resources() functions in the Tilt and enable maven docker build for the docker modules. Follow the instruction in the `Finalizing the Upgrade` to avoid duplicated docker image build.

Reviewer: A: I don't think this belongs under breaking changes since it doesn't actually break anything and they _could_ continue using their Tiltfile as-is without problems.  We should add something under _**Major Additions**_ and we could have this text live there. E.g.

```md
## Path to Production Alignment
To better align development processes with processes in CI/CD and higher environments, we no longer recommend using Tilt for building and deploying projects.  As such, upgrading projects should consider removing or at least narrowing the scope of their Tiltfile. See _**How to Upgrade**_ for more information.
```

We will likely end up tweaking this section before we actually release to align with our final recommendations, so I wouldn't stress to much about making it perfect.

>  - To improve the development cycle and docker build consistency, we have deprecated the docker_build() and local_resources() functions in the Tilt and enable maven docker build for the docker modules. Follow the instruction in the `Finalizing the Upgrade` to avoid duplicated docker image build.

Reviewer 2: ```suggestion
 - To improve the development cycle and docker build consistency, we have deprecated the `docker_build()` and `local_resources()` functions within Tilt and instead enabled the maven docker build for the docker modules. Follow the instruction in the `Finalizing the Upgrade` section to avoid duplicated docker image builds.
```

    - **NOTE:** This will update any aiSSEMBLE dependencies in 'pyproject.toml' files automatically
3. Repeat the previous step until all manual actions are resolved

> #### Maven Docker Build

Reviewer: Q/S: Why wouldn't this be under **_Precondition Steps_**? Historically this finalization section has been very light so I'm not sure many projects will be trained to pay attention to this.

Author: Yeah.. I can move it up coz original thought was this is not a required step

> To avoid duplicate docker build, remove all the related docker_build() and local_resources() function in the Tilt file.

Reviewer 2: ```suggestion
To avoid duplicate docker builds, remove all the related `docker_build()` and `local_resources()` functions from your Tiltfile.
```
